classes/class-cpt-wplf-form.php

```diff
@@ -460,6 +466,138 @@ class="code"
+
+    if ( $template_content === null ) {
+      return;
+    }
+
+    // Clear unwanted form tags. WPLF will insert those by itself when rendering a form.
+    $template_content = preg_replace( '%<form ?[^>]*?>%', '', $template_content );
+    $template_content = preg_replace( '%</form>%', '', $template_content );
+
+    $this->override_form_template( $template_content, $form_id );
+  }
+
+  /**
+   * Override a form's template with an imported template file.
+   *
+   * @param string $template_content Raw HTML content to use for the form content.
+   * @param int $form_id ID of form we're overriding the template for.
+   *
+   * @return void
+   */
+  protected function override_form_template( $template_content, $form_id ) {
+    $this->maybe_persist_override_template( $template_content, $form_id );
+
+    static $times_content_replaced = 0;
+
+    // Make the editor textarea uneditable.
+    add_filter( 'the_editor', function ( $editor ) {
+      if ( ! preg_match( '%id="wp-content-editor-container"%', $editor ) ) {
+        return $editor;
+      }
+
+      $editor = preg_replace( '%\<textarea %', '<textarea readonly="readonly" ', $editor );
+
+      $notice = _x(
+        'This form template is being overridden by code, you must edit it in your project code',
+        'Template override notice in form edit admin view',
+        'wp-libre-form'
+      );
+
+      $notice = sprintf( '<div class="wplf-template-override-notice">%s</div>', $notice );
+
+      return $notice . $editor;
+    } );
+
+    // Custom settings for the form editor.
+    add_filter( 'wp_editor_settings', function ( $settings, $editor_id ) {
+      if ( $editor_id !== 'content' ) {
+          return $settings;
+      }
+
+      $settings['tinymce'] = false;
+      $settings['quicktags'] = false;
+      $settings['media_buttons'] = false;
+
+      return $settings;
+    }, 10, 2 );
+
+    // Replace all editor content with template content.
+    add_filter( 'the_editor_content', function ( $content ) use ( $template_content, &$times_content_replaced ) {
+      // This is hacky, yes. We only want to override the content for the first
+      // editor field we come by, meaning 99% of the time we hit the wanted form
+      // template editor field at the top of the edit view page.
+      if ( $times_content_replaced > 0 ) {
+        return $content;
+      }
+
+      $times_content_replaced++;
+
+      return $template_content;
+    } );
+  }
+
+  /**
+   * Check if we need to auto-persist the form template override into WP database.
+   *
+   * @param string $template Template to maybe persist.
+   * @param int $form_id Form ID to persist template for.
+   * @param bool $force Force a persist even though not required?
+   *
+   * @return void
+   */
+  protected function maybe_persist_override_template( $template, $form_id, $force = false ) {
+    $hash_transient = 'form_tmpl_hash_' . $form_id;
```

Comment:
Maybe we should prefix this one with wplf_, just to be sure.

Reply:
Good point, will get that into place.